front/app/components/UI/QuillEditedContent/QuillEditedContent.test.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { render, screen, waitFor } from 'utils/testUtils/rtl';
+import QuillEditedContent from '.';
+import 'jest-styled-components';
+
+describe('QuillEditedContent component', () => {
+  it('Should change tabIndex on children properly', async () => {
```

Comment:
Does the test need to be async?

Reply:
This is necessary for the `waitFor` to work, which I believe makes the tests more stable? Is there a reason to make it all synchronous, or is that otherwise unnecessary in this case?

Reply:
I understand but why do you need `waitFor` in this case - nothing async is happening as far as I can tell and the test works perfectly well without it. I'm not sure I understand why it would make the test more stable.

Reply:
I got into the habit of `waitFor` to prevent flaky tests, though I guess in this case it's unnecessary if `rerender` is synchronous and the component will be always be fully updated by `rerender` before any subsequent code is called. I can change it to remove the async nature.